scripts/sign.js

```diff
@@ -0,0 +1,111 @@
+//@ts-check
+import StyleHandler from './library/style.js';
+import InputHandler from './library/input.js';
+import KeyHandler from './library/keyEvent.js';
+import DOMHandler from './library/DOM.js';
+import {
+  DEFAULT_BORDER_COLOR,
+  EMAIL_REGEX,
+  EMAIL_MESSAGE,
+  ERROR_BORDER_COLOR,
+  FOCUS_BORDER_COLOR,
+  SHOW_PASSWORD_ICON,
+  USERS,
+  PASSWORD_MESSAGE,
+  loginAction
+} from './config/signConfig.js';
+
+const loginEmailQuery = '.login__input--email';
+const loginPasswordQuery = '.login__input--password';
+const emailErrorQuery = '.email__error';
+const passwordErrorQuery = '.password__error';
+const loginBtnQuery = '.login__btn';
+
+/** @type {HTMLInputElement | null}*/
+const loginEmail = document.querySelector(loginEmailQuery);
+/** @type {HTMLInputElement | null}*/
+const loginPassword = document.querySelector(loginPasswordQuery);
+/** @type {HTMLButtonElement | null}*/
+const loginBtn = document.querySelector(loginBtnQuery);
+/** @type {HTMLImageElement | null}*/
+const eyeImg = document.querySelector('.eye-img');
+
+const handleEmailFocusIn = () => {
+  StyleHandler.display(emailErrorQuery, 'none');
+  StyleHandler.borderColor(loginEmailQuery, FOCUS_BORDER_COLOR);
+};
+
+const handleEmailFocusOut = () => {
+  if (InputHandler.isMatchRegEx(loginEmailQuery, EMAIL_REGEX))
+    return StyleHandler.borderColor(loginEmailQuery, DEFAULT_BORDER_COLOR);
+  StyleHandler.display(emailErrorQuery, 'block');
+  StyleHandler.borderColor(loginEmailQuery, ERROR_BORDER_COLOR);
+  if (InputHandler.isEmptyValue(loginEmailQuery)) DOMHandler.changeValue(emailErrorQuery, EMAIL_MESSAGE.empty);
+  else DOMHandler.changeValue(emailErrorQuery, EMAIL_MESSAGE.invalid);
```

Comment:
if-else 구문에서도 위에서 말씀드렸던 early return을 사용해보시면 좋습니다ㅎㅎ
```suggestion
  if (InputHandler.isEmptyValue(loginEmailQuery)){
     DOMHandler.changeValue(emailErrorQuery, EMAIL_MESSAGE.empty);
     reurn;
  }
 DOMHandler.changeValue(emailErrorQuery, EMAIL_MESSAGE.invalid);

Reply:
ㅋㅋㅋ 이것도 처음에 제안해주신데로 작성했었는데 코드 수정하면서 else가 있어야 알아보기 쉬운가? 하고 변경했었었는데 쿠ㅡㅜㅜㅜㅜㅜㅜ 확실히 기억해서 다음부턴 잘화용해 보겠습니다! 정정하겠습니다.https://github.com/codeit-bootcamp-frontend/4-Weekly-Mission/pull/175/commits/5b861f45f78483c96e58ee20222b52d5115d3dd4

Reply:
클린코드 법칙 상 else 나 elseif 를 지양하라는게 있긴해요ㅋㅋㅋ그래서 일반적인 클린코드관점에서 봤을때 말씀드린거고,
그런데 개발자에 따라서 else, elseif 를 선호하시는분도 계시긴합니다ㅎㅎ
준영님이 보셨을 때 else,elseif 를 붙인거 오히려 더 가독성이 좋아지면 그렇게 써도 무방할 것 같아요. 일반적인게 무조건 옳은건 아니니까요~